controllers/configmap_controller.go

```diff
@@ -121,19 +122,46 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 	}
 
 	// Fetch the ConfigMap. The predicate will have filtered out any ConfigMaps that we should not reconcile
-	// so it is safe to assume that all ConfigMaps being reconciled describe hosts that need to be present in the
-	// cluster. This also handles the case when the reconciliation is kicked off by the InstanceConfigMap being deleted.
-	// In the deletion case, an empty InstanceConfigMap will be reconciled now resulting in all existing BYOH nodes
-	// being deleted.
+	// so it is safe to assume that all ConfigMaps being reconciled are one of:
+	// 1. windows-instances, describing hosts that need to be present in the cluster.
+	// 2. windows-services, describing expected configuration of WMCO-managed services on all Windows instances
 	configMap := &core.ConfigMap{}
 	if err := r.client.Get(ctx, req.NamespacedName, configMap); err != nil {
 		if !k8sapierrors.IsNotFound(err) {
 			// Error reading the object - requeue the request.
 			return ctrl.Result{}, err
 		}
+		if req.NamespacedName.Name == servicescm.Name {
+			// Create and retrieve the services ConfigMap as it is not present
+			if configMap, err = r.createServicesConfigMap(ctx, req.NamespacedName); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
 	}
 
-	return ctrl.Result{}, r.reconcileNodes(ctx, configMap)
+	// At this point configMap will be set properly
+	switch req.NamespacedName.Name {
+	case servicescm.Name:
+		// If a ConfigMap with invalid values is found, WMCO will delete and recreate it with proper values
+		if _, err := servicescm.Parse(configMap.Data); err != nil {
+			// Deleting will trigger an event for the configmap_controller, which will re-create a proper ConfigMap
+			if err = r.client.Delete(ctx, configMap); err != nil {
+				return ctrl.Result{}, err
+			}
+			r.log.Info("Deleted invalid resource", "ConfigMap",
+				kubeTypes.NamespacedName{Namespace: req.Namespace, Name: req.Name}, "Error", err.Error())
+			return ctrl.Result{}, errors.Wrap(err, "Invalid ConfigMap schema")
+		}
```

Comment:
This block can be modularized into a function. e.g.`r.reconcileWindowsServices(ctx, configMap)` where the `switch` will read:

```
	switch req.NamespacedName.Name {
	case servicescm.Name:
              return ctrl.Result{}, r.reconcileWindowsServices(ctx, configMap)

        ...
```